ksqldb-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java

```diff
@@ -104,7 +106,15 @@ private boolean inUdfExecution() {
    * item in the class array.
    * @return true if caller is allowed
    */
+  @SuppressWarnings("checkstyle:BooleanExpressionComplexity")
   private boolean validateCaller() {
-    return getClassContext()[2].equals(PluggableUdf.class);
+    final Class caller = getClassContext()[2];
+    return caller.equals(PluggableUdf.class)
+        || caller.equals(FunctionLoaderUtils.class)
+        || caller.equals(UdfLoader.class)
+        || caller.equals(BaseAggregateFunction.class)
+        || caller.getName().equals("io.confluent.ksql.function.UdafFactoryInvoker")
+        || caller.getName().equals("io.confluent.ksql.function.UdafAggregateFunction")
+        || caller.getName().equals("io.confluent.ksql.function.UdafTableAggregateFunction");
```

Comment:
I ended up doing this since those classes are not in package which `ksqldb-engine` depends on.:)

I'm tempted to remove this function / the guard on the caller.  I am not sure why it is needed in the first place.

Reply:
I could add them without import issues. The `UdafFactoryInvoker` must be public, though.

I see them in the same package too.